services/processor/native/adapter/native_compiler.go

```diff
@@ -71,7 +71,7 @@ func getHashOfCode(code string) string {
 
 func writeSourceCodeToDisk(filenamePrefix string, code string, artifactsPath string) (string, error) {
 	dir := filepath.Join(artifactsPath, SOURCE_CODE_PATH)
-	err := os.MkdirAll(dir, 0700)
+	err := os.MkdirAll(dir, 0755)
```

Comment:
Why do we need this change? I don't see a reason that anyone other than the caller should have access to the directory.